lib/_http_incoming.js

```diff
@@ -315,6 +315,8 @@ IncomingMessage.prototype._dump = function _dump() {
   if (!this._dumped) {
     this._dumped = true;
     this.resume();
+    // Flush the existing buffered data
+    this._readableState._flush();
```

Comment:
I'm ok if you move the code you put in `_flush()` here, if it is needed help solve a bug.

At the same time, it's better to open a feature request for a new method in the stream API.